woocommerce/woocommerce-android
# Generate variations from new attributes screen

Closes: #8111

### Why
After creating attributes for a new variable product, we wanted to add the generate all variation option along with the generate random variation that was already present on the screen.

### Description
This PR adds the necessary changes to support generate-all-variation and generate-single-variation from the success attributes created screen.

### Testing instructions
Follow these steps to reach the Attributes created screen
1. Add a new variable product or open a variable product without attributes or variations
2. Tap on Add variations
3. Tap on an attribute
4. Tap on several attribute options
5. Tap on next on the top of the screen once you are happy with the options added
6. Tap on next again to reach the Attributes created screen

TC1
1. On the Attributes created screen, tap on generate variation
2. Check that a Bottom sheet opens with the options 1) Add new variation, 2) Generate all variations
3. Tap on Generate all variations
4. Check that the generate all variation dialog is shown
5. Tap on Ok
6. Check that if the operation succeeded, the new variations are added to the product

TC2
1. On the Attributes created screen, tap on generate variation
2. Check that a Bottom sheet opens with the options 1) Add new variation, 2) Generate all variations
3. Tap on Generate new variation
4. Check that if the operation succeeded, a new variation is added to the product

TC3
1. Repeat TC1 and TC2 but remove the internet connection this time before generating the variations
2. Check that an error message is shown

TC4
1. Try to generate more than 100 variations
2. Check that an error message is shown

### Images/gif

https://user-images.githubusercontent.com/18119390/211961500-67155fbe-5195-441a-9ad7-06574739baab.mp4
